### PR TITLE
Fixup Flyout Template Parts

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

- Overhaul the flyout APIs to work with the new MAUI layout expectations. 
- Add better platform level test coverage for coverage that used to be in Xamarin.UITest

### Issues Fixed

Fixes #6061

### CODE REVIEW

File: src/Controls/samples/Controls.Sample.Sandbox/SandboxShell.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System.Diagnostics;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Maui;
+using Microsoft.Maui.Controls;
+
+namespace Maui.Controls.Sample
+{
+	public partial class SandboxShell : Shell
+	{
+		public SandboxShell()
```
Comment (PureWeen) [inline src/Controls/samples/Controls.Sample.Sandbox/SandboxShell.xaml.cs:12]: Nobody worry. This is just adding the file for easier testing purposes. I haven't changed the sandbox to default to shell

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs [hunk 2]
```diff
@@ -68,6 +71,14 @@ void OnFlyoutStateChanging(object sender, AndroidX.DrawerLayout.Widget.DrawerLay
 			}
 		}
 
+		class ThingBehavior : AppBarLayout.Behavior
```
Comment (rachelkang) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:74]: What exactly is this? And why is it called `ThingBehavior`?

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutRenderer.cs [hunk 3]
```diff
@@ -124,7 +124,7 @@ void IFlyoutBehaviorObserver.OnFlyoutBehaviorChanged(FlyoutBehavior behavior)
 		int _flyoutWidthDefault;
 		double _flyoutWidth = -1;
 		double _flyoutHeight;
-		bool _flyoutFirstDrawPassFinished;
+		internal bool FlyoutFirstDrawPassFinished { get; private set; }
```
Comment (rachelkang) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutRenderer.cs:127]: Why was this change made?

File: src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Android.cs [hunk 4]
```diff
@@ -142,24 +148,118 @@ public async Task FlyoutFooterRenderersWithDefaultFlyoutItems()
...
+				var flyoutContainer = GetFlyoutMenuReyclerView(handler);
 
 				Assert.True(flyoutContainer.MeasuredWidth > 0);
 				Assert.True(flyoutContainer.MeasuredHeight > 0);
 			});
 		}
 
-		protected Task OpenDrawerLayout(ShellRenderer shellRenderer, TimeSpan? timeOut = null)
+		[Fact]
+		public async Task FlyoutItemsRendererWhenFlyoutHeaderIsSet()
```
Comment (rachelkang) [inline src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Android.cs:161]:
```suggestion
		public async Task FlyoutItemsRenderWhenFlyoutHeaderIsSet()
```

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutRenderer.cs (hunk 3 above)
Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutRenderer.cs:127]: This is used by the unit tests to validate a certain point in the life cycle

https://github.com/dotnet/maui/pull/8022/files/43b2c4d6f6d70ff46432548abba53ebe8526224d#diff-682cc392a19d37f2fba78a5128ff76b224290d624f02e9e1020eaa2a882ad9d1R249

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs (hunk 2 above)
Comment (PureWeen) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs]: Whoops....

That was when I was testing some nested scroll behaviors
Deleted this

File: src/TestUtils/src/DeviceTests/AssertionExtensions.cs [hunk 5]
```diff
@@ -42,5 +42,14 @@ public static void AssertWithMessage(Action assertion, string message)
 				Assert.True(false, $"Message: {message} Failure: {e}");
 			}
 		}
+
+		public static void CloseEnough(double expected, double actual, double epsilon = 0.2, string? message = null)
```
Comment (rachelkang) [inline src/TestUtils/src/DeviceTests/AssertionExtensions.cs:46]: Why is it that we can't get equal but only "close enough" with the various flyout frame measurements?

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutRenderer.cs (hunk 3 above)
Comment (rachelkang) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutRenderer.cs:127]: just trying to understand the change to `internal` and `private set` better - why would the original bool not have sufficed?

**Review by rachelkang — APPROVED** [review]
LGTM :) and validated this indeed fixes #6061

File: src/TestUtils/src/DeviceTests/AssertionExtensions.cs (hunk 5 above)
Comment (PureWeen) [reply src/TestUtils/src/DeviceTests/AssertionExtensions.cs:46]: The conversion of our DP values to pixels based on the density of the device and then back to DP will result in some fractional precision loss. A 0.2 epsilon represents less than half a pixel.

**Pull request merged**